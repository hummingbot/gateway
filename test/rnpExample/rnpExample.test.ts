import {
  useABC,
  useUnmappedDep,
  useDTwice,
  useProtoDep,
  useUnmappedMethodMocked,
} from './rnpExample.api-test-cases';
import { RnpExampleTestHarness } from './rnpExample.test-harness';

describe('RnpExample', () => {
  let harness: RnpExampleTestHarness;

  beforeAll(async () => {
    harness = new RnpExampleTestHarness();
    await harness.setupMockedTests();
  });

  afterEach(async () => {
    await harness.reset();
  });

  afterAll(async () => {
    await harness.teardown();
  });

  it('useABC', async () => {
    await useABC.processPlayRequest(harness);
  });

  it('useDTwice', async () => {
    await useDTwice.processPlayRequest(harness);
  });

  it('useUnmappedMethodRecorder', async () => {
    // Used to force snapshot file to match exactly
    expect({
      unmapped: expect.any(String),
    }).toMatchSnapshot();
  });

  it('useUnmappedMethodMocked', async () => {
    await useUnmappedMethodMocked.processPlayRequest(harness);
  });

  it('useProtoDep', async () => {
    await useProtoDep.processPlayRequest(harness);
  });

  it('useUnmappedDep', async () => {
    await useUnmappedDep.processPlayRequest(harness);
  });
});
