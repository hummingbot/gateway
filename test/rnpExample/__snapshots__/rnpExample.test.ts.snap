// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RnpExample useABC 1`] = `
{
  "a": "real methodA-0.3167918897895934",
  "b": "real methodB-0.5959480847973675",
  "c": Any<String>,
}
`;

exports[`RnpExample useDTwice 1`] = `
{
  "d1": "real methodD-0.9822143426913119",
  "d2": "real methodD-0.3523185054710709",
}
`;

exports[`RnpExample useDep2 1`] = `
{
  "z": Any<String>,
}
`;

exports[`RnpExample useUnmappedMethodMocked 1`] = `
{
  "error": "FailedDependencyError",
  "message": "Failed to useUnmappedMethod: Unmocked method was called: dep1_A.methodUnmapped",
  "statusCode": 424,
}
`;

exports[`RnpExample useUnmappedMethodRecorder 1`] = `
{
  "unmapped": "real methodUnmapped-0.14017814520237115",
}
`;
